modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/maven/aether/KobaltMavenResolver.kt

```diff
@@ -38,6 +38,14 @@ class KobaltMavenResolver @Inject constructor(val settings: KobaltSettings,
     fun resolve(id: String, scope: Scope? = null, filter: DependencyFilter? = null): DependencyResult {
         val dependencyRequest = DependencyRequest(createCollectRequest(id, scope), filter)
         val result = system.resolveDependencies(session, dependencyRequest)
+        if (args.downloadSources) {
+            listOf("sources", "javadoc").forEach {
+                val artifact = DefaultArtifact(id)
+                val sourceArtifact = DefaultArtifact(artifact.groupId, artifact.artifactId, it, artifact.extension,
+                        artifact.version)
+                system.resolveDependencies(session, DependencyRequest(createCollectRequest(sourceArtifact, scope), filter))
```

Comment:
Why this change? By doing this in `createCollectRequest()`, I actually only send one resolution to MavenResolver, and it will download everything in one go. Here, you are doing three separate `resolveDependencies` (one for the regular jar, one for the sources and one for the javadocs).